### ISSUE TITLE
PHP Notice: Undefined index: in 2.3.0/CodeSniffer/File.php

### ISSUE THREAD

**Issue by remcotolsma** [squizlabs/PHP_CodeSniffer#521]
I'm getting PHP notices with the following command and file:

```
phpcs --standard=PHPCS test.php
```

```php
<?php while ( have_posts() ) : the_post(); ?>

	<?php if ( true ) : ?>

		<?php

		if ( true ) {
			
		} else {
			
		}

		?>

	<?php endif; ?>

<?php endwhile; ?>
```

```
Notice: Undefined index:  in /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer/File.php on line 3411

Call Stack:
    0.0002     241624   1. {main}() /usr/local/Cellar/php-code-sniffer/2.3.0/scripts/phpcs:0
    0.0075    1738704   2. PHP_CodeSniffer_CLI->runphpcs() /usr/local/Cellar/php-code-sniffer/2.3.0/scripts/phpcs:25
    0.0078    1804416   3. PHP_CodeSniffer_CLI->process() /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer/CLI.php:95
    0.4402    5058992   4. PHP_CodeSniffer->processFiles() /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer/CLI.php:867
    0.4403    5062088   5. PHP_CodeSniffer->processFile() /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer.php:619
    0.4404    5063024   6. PHP_CodeSniffer->_processFile() /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer.php:1679
    0.4408    5108576   7. PHP_CodeSniffer_File->start() /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer.php:1801
    0.4422    5355536   8. Generic_Sniffs_WhiteSpace_ScopeIndentSniff->process() /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer/File.php:567
    0.4435    5417592   9. PHP_CodeSniffer_File->findFirstOnLine() /usr/local/Cellar/php-code-sniffer/2.3.0/CodeSniffer/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php:409
```

https://github.com/squizlabs/PHP_CodeSniffer/blob/2.3.0/CodeSniffer/File.php#L3411

**Comment by aik099** [comment]
I see a lot of issues related to this PHP short syntax. I'm wondering why it's so attractive.

**Comment by gsherwood** [comment]
Sorry it's taken me a while to get to this. I've just done some testing on this and confirmed the problem exists in the current stable 2.3.0 but has been fixed in the master branch already, so this fix will be released in 2.3.1.

In the meantime, you can grab the latest master code and install that directly if you need it.
